src/libcrun/cgroup.c

```diff
@@ -494,6 +494,10 @@ int systemd_finalize (oci_container_linux_resources *resources, int cgroup_mode,
       subpath = strchr (subsystem, ':') + 1;
       *(subpath - 1) = '\0';
 
+      /* Only process cgroupv2, which will have no listed subsystem */
+      if (strcmp(subsystem, ""))
```

Comment:
The code here runs also on cgroups v1 (as per the condition at line 487, it is skipped only when cgroups != unified v2 and it is running with euid != 0).
I think the code is correct, as we probably don't want to join named hierarchies we don't control also on cgroups v1, but the comment is a bit misleading.  Could it be replaced with something saying it skips named hierarchies?